cachi2/core/package_managers/yarn/project.py

```diff
@@ -120,9 +148,12 @@ def yarn_cache(self) -> RootedPath:
         The cache location is affected by the cacheFolder configuration in yarnrc. See:
         https://v3.yarnpkg.com/configuration/yarnrc#cacheFolder.
         """
-        return NotImplemented
+        return self.source_dir.join_within_root(self.yarn_rc.cache_path)
 
     @classmethod
     def from_source_dir(cls, source_dir: RootedPath) -> "Project":
         """Create a Project from a sources directory path."""
-        return NotImplemented
+        yarn_rc = YarnRc.from_file(source_dir.join_within_root(".yarnrc.yml"))
```

Comment:
The yarnrc is not mandatory, so we need to account for cases in which it's absent. Calling `yarn init` in a folder does not autogenerate it.

Reply:
What if package.json is missing? Are we going to raise a `PackageRejected`?

Reply:
> Calling yarn init in a folder does not autogenerate it.

Calling `yarn init -2` does, but I think you're right that it's not mandatory

Reply:
Raising `PackageRejected` in PackageJson classmethod.